test/framework/integration_framework/fake_peer/behaviour/behaviour.cpp

```diff
@@ -25,25 +25,37 @@ namespace integration_framework {
       subscriptions_.emplace_back(
           getFakePeer().getMstStatesObservable().subscribe(
               [this](const auto &message) {
-                this->processMstMessage(message);
+                if (auto alive = fake_peer_wptr_.lock()) {
```

Comment:
Will `this` object outlive the observable? If there are no such guarantees, probably `fake_peer_wptr_` has to be captured by value.
Same below.

Reply:
great catch, thanks!